Add module documentation for the ChapelRange internal module

daviditen opened the pull request:
This is a first pass over documenting the ChapelRange internal module. It
could probably use some cleanup and further attention, but I think it is
pretty good for a first pass.

Added internal/ChapelRange.chpl to the modules/Makefile.

ronawho commented:
Overall, I think this is a good first pass!

I think your inline code markup needs to use 2 `'s instead of 1

Lets hold off on committing this just yet. We're still working through some things for generating internal documentation. In particular for most of the internal modules we found we needed to do some post-processing on the .rst to rename internal types. In that post-processing we also have stuff to rename modules, and to strip out internal `chpl_*` functions.

https://github.com/chapel-lang/chapel/pull/2560/files introduces the post-processing stuff, so we should wait until after that is in and see if we need to make any changes

Oh, also thanks for fixing some of my old spelling errors!

ronawho commented on modules/Makefile:
```diff
@@ -52,6 +52,7 @@ $(SYS_CTYPES_MODULE_DOC): $(MAKE_SYS_BASIC_TYPES)
 
 
 MODULES_TO_DOCUMENT = \
+	internal/ChapelRange.chpl \
```
Now that Ben's change is in, you'll want to add this to `INTERNAL_MODULES_TO_DOCUMENT` instead

ronawho commented on modules/internal/ChapelRange.chpl:
```diff
@@ -1747,10 +1977,12 @@ module ChapelRange {
   { return chpl__extendedEuclidHelper(u,v); }
-  
+
+  pragma "no doc"
```
I think it'd be good to add an entry to the `modules/internal/fixInternalDocs.sh` script for Ranges and use the `removePrefixFunctions` to avoid having to no doc all these internal functions. Long term, chpldoc should know not to document things starting with `chpl_`

ronawho replied:
Note that it also removes any functions that don't start with a letter so you can also remove the `no docs` from the overloaded methods.

ronawho commented:
It'd also be nice to use the `fixTitle` function to replace `module::ChapelRange` with just `Range`.

I think the syntax for that is: `fixTitle "Range" $file`

vasslitvinov commented on modules/internal/ChapelRange.chpl:
```diff
@@ -16,21 +16,98 @@
  * limitations under the License.
  */
+/*
+  The ChapelRange module provides the implementation for the built in range
```
I suggest removing the first sentence "The ChapelRange module provides the implementation ..." because it is our internal implementation detail. I envision when users click on "Range" in the contents, they get to a page that talks about ranges regardless of whether there is a module for it or not. Ditto tuples, arrays, etc.

ronawho replied:
Agreed, this sentence can be removed.

@vasslitvinov The `fixTitle` helper routine will take care of renaming "module::ChapelRange" to just "Range". It was added to do exactly as you're saying: abstract the actual implementation details and just refer to the type.

daviditen commented:
I've taken care of the comments here now.

* Updated fixInternalDocs.sh to handle ChapelRange

  * Rename from "moudle::ChapelRange" to "Ranges"
  * Remove functions that should be ignored for documentation.

* Removed "no doc" from things starting with "chpl_" and operators and non-letters.

* Removed the first sentence "This module implements ..."

ronawho commented:
This looks in good shape to me. The only thing I see is that I would no-doc `isRangeType` and `isRangeValue` and just rely on the already documented `isRange`. Let me check with Brad on this though

ronawho commented:
Looks like we are documenting `is*Type`. Can you add a no-doc to isRangeValue? Other than that, looks good to merge
